app/controllers/repositories_controller.rb

```diff
@@ -1,9 +1,13 @@
 class RepositoriesController < SessionsController
-  before_action :current_user
+  before_action :current_user, :check_session
   before_action :signed_in, except: [:index, :show, :download, :download_files]
-  before_action :set_repository, only: [:show, :add_like, :destroy, :edit, :update, :download_files]
+  before_action :set_repository, only: [:show, :add_like, :destroy, :edit, :update, :download_files, :check_auth, :pass_authenticate]
+  before_action :check_auth, only: [:show]
```

Comment:
@tahnok is it okay if I limited :check_auth to 'show'?
Also I had to move it after set_repository cus the repository needs to exist to check the conditions.

Reply:
Ya, that makes sense